docs/data/system/components/grid/grid.md

```diff
@@ -0,0 +1,148 @@
+
+{{"demo": "SpacingGrid.js", "bg": true, "hideToolbar": true}}
+
+### Row & column spacing
+
+The `rowSpacing` and `columnSpacing` props allow for specifying the row and column gaps independently.
+It's similar to the `row-gap` and `column-gap` properties of [CSS Grid](/system/grid/#row-gap-amp-column-gap).
+
+{{"demo": "RowAndColumnSpacing.js", "bg": true}}
+
+## Responsive values
+
+You can switch the props' value based on the active breakpoint.
+For instance, we can implement the ["recommended"](https://material.io/design/layout/responsive-layout-grid.html) responsive layout grid of Material Design.
+
+{{"demo": "ResponsiveGrid.js", "bg": true}}
+
+Responsive values is supported by:
+
+- `columns`
+- `columnSpacing`
+- `direction`
+- `rowSpacing`
+- `spacing`
+- all the [other props](#system-props) of the system
+
+## Auto-layout
+
+The Auto-layout makes the _items_ equitably share the available space.
+That also means you can set the width of one _item_ and the others will automatically resize around it.
+
+{{"demo": "AutoGrid.js", "bg": true}}
+
+### Variable width content
+
+Set one of the size breakpoint props to `"auto"` instead of `true` / a `number` to size
+a column based on the natural width of its content.
+
+{{"demo": "VariableWidthGrid.js", "bg": true}}
+
+## Nested Grid
+
+The grid container that renders inside another grid container is a nested grid which inherits the `columns` and `spacing` from the top. The deep nested grid will inherit the props from the upper nested grid if it receives those props.
+
+👌 There is no limit on the levels of the nested grids but you should use it wisely.
+
+{{"demo": "NestedGrid.js", "bg": true}}
+
+## Columns
+
+You can change the default number of columns (12) with the `columns` prop.
+
+{{"demo": "ColumnsGrid.js", "bg": true}}
+
+## Custom breakpoints
+
+If you specify custom breakpoints to the theme, you can use those names as grid item props and in responsive values. For **Typescript**, you have to set module augmentation on theme breakpoints and the grid breakpoints interfaces.
+
+{{"demo": "CustomBreakpointsGrid.js", "bg": true}}
+
+## Offset
+
+Move the item to the right by using offset props which can be a number or `"auto"`.
+
+- a number, for example, `mdOffset={2}` moves the item to the right by 2 columns starts from `md` breakpoint and up.
+- The `"auto"` value moves the item to the right edge of the grid container.
+
+{{"demo": "OffsetGrid.js", "bg": true}}
+
+## Limitations
+
+### Negative margin
+
+The spacing between items is implemented with a negative margin. This might lead to unexpected behaviors. For instance, it can cause an overflow on mobile viewport (horizontal scroll), you need to apply `overflow: hidden;` to the parent of the grid container.
+
+```js
+<Box sx={{ maxWidth: '100vw', overflow: 'hidden' }}>
+  <Grid container spacing={4}>
+    ...grid items
+  </Grid>
+</Box>
+```
+
+### `direction: column | column-reverse`
```

Comment:
Using code in a title looks awkward. As far as I remember we decided not to do this, but maybe I am wrong. cc @samuelsycamore

Reply:
Updated.